neo/Prompt/Commands/Search.py

```diff
@@ -0,0 +1,75 @@
+from neo.Prompt.CommandBase import CommandBase, CommandDesc, ParameterDesc
+from neo.Prompt.PromptData import PromptData
+from neo.Prompt.Utils import get_arg
+from neo.Core.Blockchain import Blockchain
+from prompt_toolkit.shortcuts import print_formatted_text
+from prompt_toolkit.formatted_text import FormattedText
+from neo.logging import log_manager
+import json
+
+
+logger = log_manager.getLogger()
+
+
+class CommandSearch(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+        self.register_sub_command(CommandSearchAsset())
+        self.register_sub_command(CommandSearchContract())
+
+    def command_desc(self):
+        return CommandDesc('search', 'search the blockchain')
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+
+        if not item:
+            print("run `search help` to see supported queries")
+            return
+
+        try:
+            return self.execute_sub_command(item, arguments[1:])
+        except KeyError:
+            print(f"search: {item} is an invalid parameter")
+            return
+
+
+class CommandSearchAsset(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+
+        results = Blockchain.Default().SearchAssetState(item)
+        print("Found %s results for %s" % (len(results), item))
+        for asset in results:
+            bjson = json.dumps(asset.ToJson(), indent=4)
+            tokens = [("class:number", bjson)]
+            print_formatted_text(FormattedText(tokens), style=PromptData.Prompt.token_style)
+        return results
+
+    def command_desc(self):
+        p1 = ParameterDesc('query', 'supports name, issuer, or admin searches')

```

Comment:
I included this example to highlight this issue. I am wondering if this is operating as intended because the original name was "AntShare" as it is saved and only rebranded to "NEO". How would you like me to proceed?

Reply:
I think that's an OK solution. I don't expect any big list of new assets (if at all) and even less that will rename.